campaign-launcher/interfaceV2/src/api/recordingApiClient.ts

```diff
@@ -0,0 +1,180 @@
+import axios, { 
+  AxiosError, 
+  AxiosInstance, 
+  AxiosRequestConfig, 
+  AxiosRequestHeaders, 
+  AxiosResponse, 
+  InternalAxiosRequestConfig, 
+  Method
+} from 'axios';
+
+import { TokenData, TokenManager } from "../utils/TokenManager";
+
+type RefreshPromise = Promise<AxiosResponse<TokenData>>;
+
+type RecordingApiClientConfig = {
+  baseUrl: string;
+  headers?: AxiosRequestHeaders;
+  tokenManager: TokenManager;
+};
+
+export const REFRESH_FAILURE_EVENT = 'refresh-failure';
+
+class HttpError extends Error {
+  public readonly status: number;
+
+  constructor(message: string, status: number) {
+    super(message);
+    this.status = status;
+  }
+
+  static fromAxiosError(error: AxiosError): HttpError {
+    const status = error.response?.status || 0;
+    const message = error.message || 'Request failed';
+    return new HttpError(message, status);
+  }
+} 
```

Comment:
Would be nice to have some `BaseError` class to make sure `name` property is properly set
```
class BaseError extends Error {
  constructor(message: string) {
    super(message);
    this.name = this.constructor.name;
  }
}

```


```suggestion
class HttpError extends BaseError {
  constructor(message: string, readonly status?: number) {
    super(message);
  }

  static fromAxiosError(error: AxiosError): HttpError {
    return new HttpError(
      error.message || 'Request failed',
      error.response?.status,
    );
  }
}
```